dtm0/recovery.c

```diff
@@ -1834,24 +1836,14 @@ static void local_recovery_fom_coro(struct m0_fom *fom)
 	 * REDOs.
 	 * mkfs does not require DTM0 support as well.
 	 */
-	F(recovered) = rf->rf_is_volatile || is_mkfs(fom);
-
-	if (!F(recovered)) {
-		/* Wait until the moment where we should start recovery. */
-		do {
-			M0_CO_FUN(CO(fom), heq_await(fom, &F(state), &F(eoq)));
-			if (F(eoq))
-				goto out;
-
-			if (F(state) == M0_NC_ONLINE && !ALL2ALL)
-				break;
-		} while (F(state) != M0_NC_DTM_RECOVERING);
+	F(recovered) = rf->rf_is_volatile;
```

Comment:
Why is_mkfs() is removed from here ?

Reply:
This fix was made after thorough discussion with IvanA.

Reasons:

1. Unit test thinks it runs in mkfs mode (`cs_ctx->cc_mkfs` is true).  So this logic was breaking recovery machine unit tests.
2. This is_mkfs() check was added "blindly", on early stages, and never completely validated.
3. ==> So, this code was added "for future" and is breaking unit tests.  We discussed and decision was -- remove it now, and add proper code "in future" when the need arises and we know exactly what is needed and for which purpose.
4. After removal of this fix, all2all test (which also runs mkfs) succeeded, with no issues, which proves that fix is not breaking anything.
5. Treating "recovery during mkfs" is definitely future, not PI7.  And it is not yet defined and designed and researched.